tests/PHPStan/Analyser/Bug10862Test.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Analyser;
+
+use PHPStan\Testing\TypeInferenceTestCase;
+use const PHP_VERSION_ID;
+
+class Bug10862Test extends TypeInferenceTestCase
+{
+
+	public function dataFileAsserts(): iterable
+	{
+		$path = 'bug-10862-php8.3';
+		if (PHP_VERSION_ID < 80300) {
+			$path = 'bug-10862-php8.0';
+		}
+		if (PHP_VERSION_ID < 80000) {
+			$path = 'bug-10862';
+		}
+		yield from self::gatherAssertTypes(__DIR__ . '/data/' . $path . '.php');
```

Comment:
I'd prefer if the conditions contained the whole `yield from` call instead of concatenating the path dynamically like this. It will make the path clickable in IDE.